### DISCUSSION TITLE
error TS2307: Cannot find module '../../../types/api-output' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by HarryLu33** [TEAMMATES/teammates#12603] [Help requests] [answered]
error TS2307: Cannot find module '../../../types/api-output' or its corresponding type declarations.
- **Environment**: 
master branch at commit 62139bc8245b6fe00012a912e653a6b183391b00
**Steps to reproduce**
Run src\web\app\components\comment-box\comment-edit-form\comment-edit-form.component.spec.ts file. 

**Expected behaviour**
Compile successfully

**Actual behaviour**
`Test suite failed to run

    src/web/app/components/teammates-common/feedback-path.pipe.ts:2:41 - error TS2307: Cannot find module '../../../types/api-output' or its corresponding type declarations.

    2 import { FeedbackParticipantType } from '../../../types/api-output';
                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~`


**Additional info**
<!-- You can enter any additional relevant info here. -->
<!-- If you know what caused the bug and/or how to resolve the bug, you can include them in this section. -->
I am currently working on issue #12588 to Improve the test code coverage of CommentEditFormComponent. When I tried to run the test, the code could not compile. It shows cannot find module '../../../types/api-output'. I am new to this project, is there anything I missed to get this error?

**Comment by jasonqiu212** [comment]
Hi @HarryLu33, Have you tried running the `./gradlew createConfigs` command to generate the config files? For more information, please refer to our [Setting Up](https://teammates.github.io/teammates/setting-up.html) documentation page :)

**Comment by HarryLu33** [comment]
Thanks for your reply!
yes, I have gone through all the setup steps. However the types/api-output directory is still missing

![Screenshot 2023-10-10 000229](https://github.com/TEAMMATES/teammates/assets/141628899/5a2aba0b-353d-4c2d-9fab-3ad6395ecdcf)
![image](https://github.com/TEAMMATES/teammates/assets/141628899/191171c8-3f4e-46a4-be85-0300aafd0217)

**Comment by cedricongjh** [comment, marked as answer]
hi @HarryLu33 I think what you should be missing is `gradLew generateTypes`, do try that out and let us know if it works!

**Reply by HarryLu33** [reply]
It works!
Thank you so much for your time and support!
